src/caffeine/caffeine.h

```diff
@@ -33,6 +33,10 @@ int caf_num_images(gex_TM_t team);
 void* caf_allocate(mspace heap, size_t bytes);
 void caf_deallocate(mspace heap, void* mem);
 
+// PRIF Queries
+
+void* caf_convert_base_addr(void* addr, int image, gex_TM_t initial_team);
```

Comment:
The Fortran declaration for this `bind(C)` function has a return value of `integer(c_intptr_t)`, but the C definition declares the return type as `void *`.

**values** of these types have convertible representations, but `intptr_t` and `void *` are not guaranteed to have the same in-memory representation (for example, `intptr_t` is permitted to be "wider" than `void *`). Consequently, the Fortran standard correctly does not guarantee these types as interoperable with each other: IIUC this is a mismatched declaration and therefore UB. I believe the C version needs to be declared to return `intptr_t` to ensure guaranteed compatibility.

IIUC neither compiler will report this class of type discrepancies between `bind(C)` Fortran function declarations and the corresponding C function definition. As such, we need to be especially vigilant with manually checking such correspondences. I suggest adding mention of this fact to the coding conventions.

Reply:
@bonachea I have added a note to the new README. The wording may be improved, or the section in which this note lives in the README may be changed.